source/mqtt-tls.rst

```diff
@@ -0,0 +1,341 @@
+
+- There is a script, see more in `CertReq`_.
+- And t there's a feature in the web interface (GUI) to generate
+  certificates easily, see more in A_SER_CRIADO.
+
+In addition, you can use `OpenSSL`_ to create
+certificates and sign using the `API - x509-identity-mgmt`_,
+see more  at `X.509 Identity Management`_.
+
+A brief explanation of how to use `CertReq`_:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As prerequisites this uses `git`_, `OpenSSL`_, `curl`_ and `jq`_ .
+
+On Debian-based Linux distributions, you can install
+these prerequisites by running:
+
+.. code-block:: bash
+
+  sudo apt install git
+  sudo apt-get install curl
+  sudo apt-get install jq
+  sudo apt-get install openssl
+
+Download `CertReq`_  on your machine directly from dojot repository and switch
+to the corresponding version of your dojot environment:
+
+.. code:: shell
+
+  git clone https://github.com/dojot/dojot.git
+  cd dojot
+  git checkout v0.5.0
+
+Enter in ``certreq`` directory:
+
+.. code:: shell
+
+  cd tools/certreq
+
+Finally, you can run the script to generate
+certificates and keys as follows:
+
+.. code:: shell
+
+      ./bin/certreq.sh \
+         -h localhost \
+         -p 8000 \
+         -i 'a1998e' \
+         -u 'admin' \
+         -s 'admin'
+
+Given a *username* ``admin`` and *password* ``admin``,
+this command will request a certificate
+with *device ID (identifier)* ``a1998e``
+for the dojot platform *host* ``localhost`` on *port* ``8000``.
+
+.. NOTE::  For more understanding about parameters above, check `CertReq - Parameters`_, and for
+            see more details about this above process access
+            `How to connect a device with the IoTAgent-VerneMQ with mutual TLS`_
+            and `CertReq`_
+
+And in the end this tool will create the directories ``./ca`` and
+``./cert_{DEVICE_ID}`` to store the certificates
+and public/private keys.
+
+
+Simulating a device with mosquitto
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the mosquitto to emulate a device to publish and
+subscribe in dojot via MQTT.
+
+Before install `mosquitto_pub` and `mosquitto_sub` from
+package `mosquitto-clients` on Debian-based Linux distributions:
+
+.. ATTENTION::
+    Some Linux distributions, Debian-based Linux distributions in particular,
+    have two packages for
+    `mosquitto`_, one containing tools to access it (i.e. mosquitto_pub and
+    mosquitto_sub for publishing messages and subscribing to topics) and
+    another one containing the MQTT broker too. In this tutorial, only the
+    tools from package `mosquitto-clients` on Debian-based Linux distributions
+    are going to be used.
+    Please check if MQTT broker is not running before starting dojot
+    (by running commands like ``ps aux | grep mosquitto``) to avoid port conflicts.
+
+On Debian-based Linux distributions you can install
+``mosquitto-clients``  running:
+
+.. code:: bash
+
+   sudo apt-get install mosquitto-clients
+
+
+Broker MQTT IotAgent VerneMQ (Default)
```

Comment:
Sounds a bit strange to me, maybe only `VerneMQ (default)` as a title would do the job.